### DISCUSSION TITLE
Scan nested object attribute in array

### DISCUSSION THREAD

**Discussion by montoyadamien** [dynamoose/dynamoose#1526] [Q&A] [answered]
Hello,

I got a question and I did not find answers in the doc or on the net.

Given the following schema:
```
const dangerSchema = new dynamoose.Schema(
    {
      id: {
        type: String,
        required: true,
        rangeKey: true,
      },
      supports: {
        type: Array,
        schema: [
          {
            type: Object,
            schema: {
              userId: {
                type: String,
                required: true,
              },
              createdAt: {
                type: Number,
                required: true,
              },
            },
          },
        ],
        default: [],
      },
    }
  );
```
How do I fetch dangers that has an userId in the object that is in the supports array ?

I tried lots ot things including this and it seems not working:
```
await Danger.scan({
        'supports.userId': {
          eq: userId,
        },
      }).exec()
    ).toJSON();
```

Do you have an idea of the request to do ?

I am using dynamoose 2.8.5.

Thanks :)

**Comment by fishcharlie** [comment]
Currently this isn't supported. Nested property support for conditionals, updates, etc isn't very well supported at all in Dynamoose. A very similar issue exists at https://github.com/dynamoose/dynamoose/issues/665 for updates.

It's on my radar tho.

**Reply by montoyadamien** [reply]
Okey, thanks for your answer ans for the work you are doing !

**Comment by fishcharlie** [comment, marked as answer]
Closing as a duplicate of https://github.com/dynamoose/dynamoose/issues/1589 to keep everything in one place.
